### PR TITLE
add InProgress ErrorKind gated behind io_error_inprogress feature

### PR DESCRIPTION
**Pull request by aviramha**
Follow up on https://github.com/rust-lang/libs-team/issues/92#issuecomment-2371666560

### CODE REVIEW

File: library/std/src/io/error.rs [hunk 1]
```diff
@@ -400,6 +400,11 @@ pub enum ErrorKind {
     #[stable(feature = "out_of_memory_error", since = "1.54.0")]
     OutOfMemory,
 
+    /// The operation was partially sucessful and needs to be checked
+    /// later on due to not blocking
+    #[unstable(feature = "io_error_inprogress", issue = "none")]
```
Comment (Noratrieb) [inline library/std/src/io/error.rs:405]: this is the only place where the attribute is needed, remove the others

Comment (aviramha) [reply library/std/src/io/error.rs:405]: Done

**Comment by Noratrieb** [comment]
nice, could you squash the commits together? you've got my approval afterwards 🐱

**Comment by aviramha** [comment]
> nice, could you squash the commits together? you've got my approval afterwards 🐱

Done :) 
Thanks for the swift review! 😸

File: library/std/src/io/error.rs [hunk 2]
```diff
@@ -400,6 +400,11 @@ pub enum ErrorKind {
     #[stable(feature = "out_of_memory_error", since = "1.54.0")]
     OutOfMemory,
 
+    /// The operation was partially sucessful and needs to be checked
```
Comment (dtolnay) [inline library/std/src/io/error.rs:403]:
```suggestion
    /// The operation was partially successful and needs to be checked
```

**Review by dtolnay — CHANGES_REQUESTED** [review]

File: library/std/src/io/error.rs [hunk 3]
```diff
@@ -400,6 +400,11 @@ pub enum ErrorKind {
     #[stable(feature = "out_of_memory_error", since = "1.54.0")]
     OutOfMemory,
 
+    /// The operation was partially successful and needs to be checked
+    /// later on due to not blocking
```
Comment (Noratrieb) [inline library/std/src/io/error.rs:404]:
```suggestion
    /// later on due to not blocking.
```
while you're already at it, sorry

**Comment by Noratrieb** [comment]
thanks!
@bors r+ rollup

**Comment by Noratrieb** [comment]
you are absolutely correct, I noticed that during review and then forgot to comment about it, thanks.
@aviramha please create a tracking issue and submit another PR updating the `issue =` number that you set to none, sorry for forgetting about it

**Comment by Noratrieb** [comment]
no, let me do that for you

**Comment by Noratrieb** [comment]
I put up  #130842

**Pull request merged**